hadoop-hdds/server-scm/src/test/java/org/apache/hadoop/hdds/scm/container/replication/health/TestRatisReplicationCheckHandler.java

```diff
@@ -363,6 +363,28 @@ public void testHandlerReturnsFalseWhenAllReplicasAreUnhealthy() {
     Assert.assertFalse(result.isReplicatedOkAfterPending());
     Assert.assertFalse(result.underReplicatedDueToDecommission());
 
+    Assert.assertFalse(healthCheck.handle(requestBuilder.build()));
+    Assert.assertEquals(0, repQueue.underReplicatedQueueSize());
+    Assert.assertEquals(0, repQueue.overReplicatedQueueSize());
+    Assert.assertEquals(0, report.getStat(
+        ReplicationManagerReport.HealthState.UNDER_REPLICATED));
```

Comment:
This and the above check are related to point 5 in the PR description right? Handling over-replicated unhealthy is still WIP?

Reply:
Handling all UNHEALTHY replicas works as it does in legacy. This is checked and queued for over/under rep in `RatisUnhealthyReplicationCheckHandler` and the required commands are created in `RatisOverReplicationHandler` and `RatisUnderReplicationHandler`.

Point 5 is about situations where we have excess UNHEALTHY and mismatched replicas. `RatisReplicationCheckHandler` won't queue for over rep as long as there are some mismatched replicas. `RatisUnhealthyReplicationCheckHandler` is however still queuing them - this is inconsistent as you pointed out in the comment below. I can fix this by either making both checks more conservative (not queuing if there are mismatched replicas) or by just checking if we have replication factor number of matching replicas and then queuing.